pkg/simple/client/ldap/ldap.go

```diff
@@ -58,24 +58,24 @@ func NewLdapClient(options *LdapOptions, stopCh <-chan struct{}) (*LdapClient, e
 	return client, nil
 }
 
-func (l *LdapClient) Ldap() ldap.Client {
-	if l.pool != nil {
-		conn, err := l.pool.Get()
-		if err != nil {
-			klog.Error(err)
-			return nil
-		}
-
-		err = conn.Bind(l.options.ManagerDN, l.options.ManagerPassword)
-		if err != nil {
-			conn.Close()
-			klog.Error(err)
-			return nil
-		}
-		return conn
+func (l *LdapClient) NewConn() (ldap.Client, error) {
+	if l.pool == nil {
+		klog.Fatalln("ldap connection pool is not initialized")
```

Comment:
not appropriate to do fatal logging, since we will check return values, I think it's better to return error instead